tests/infra/clients.py

```diff
@@ -498,7 +498,7 @@ def request(
                 timeout=timeout,
                 content=request_body,
             )
-        except httpx.ReadTimeout as exc:
+        except httpx.TimeoutException as exc:
```

Comment:
A maybe-risky change. See [here](https://www.python-httpx.org/exceptions/#the-exception-hierarchy) for httpx's exception hierarchy. This change means that other kinds of httpx timeout will also be converted to `TimeoutError`s for our infra to deal with. The motivation was seeing `ConnectTimeout`s when a TLS handshake failed (presumably because of a slow node?), which triggered an early test failure despite passing through multiple parts of the stack which were explicitly repeating/robust to delays.

Reply:
So what is the consequence of raising them as a timeout exception instead of a runtime error? Do we gracefully handle timeout exceptions somewhere else?

Reply:
Yes - `TimeoutError` (our type) is caught and handled differently in many places in the code.